src/app/charts/chart.component.ts

```diff
@@ -137,6 +158,11 @@ export class ChartComponent implements OnChanges {
         this.imageExportService.downloadAsPNG(this.chart.indicator.name, fileName);
     }
 
+    public onThresholdSelected($event) {
+        delete $event.event; // everything else is a threshold parameter
```

Comment:
Okay. This was the recommended approach from what I'd found, but it is unwieldy to have to go hang extra arguments onto an existing object. I think the thing passed does need to be of the special `event` type though.